netbox/netbox/tests/test_base_classes.py

```diff
@@ -0,0 +1,329 @@
+    PrimaryModelTable,
+)
+
+
+class FormClassesTestCase(TestCase):
+
+    @staticmethod
+    def get_form_for_model(model, prefix=''):
+        """
+        Import and return the form class for a given model.
+        """
+        app_label = model._meta.app_label
+        model_name = model.__name__
+        return import_string(f'{app_label}.forms.{model_name}{prefix}Form')
+
+    @staticmethod
+    def get_model_form_base_class(model):
+        """
+        Return the base form class for creating/editing the given model.
+        """
+        if model._meta.app_label == 'dummy_plugin':
+            return
+        if issubclass(model, PrimaryModel):
+            return PrimaryModelForm
+        if issubclass(model, OrganizationalModel):
+            return OrganizationalModelForm
+        if issubclass(model, NestedGroupModel):
+            return NestedGroupModelForm
+        if issubclass(model, NetBoxModel):
+            return NetBoxModelForm
+
+    @staticmethod
+    def get_bulk_edit_form_base_class(model):
+        """
+        Return the base form class for bulk editing the given model.
+        """
+        if model._meta.app_label == 'dummy_plugin':
+            return
+        if issubclass(model, PrimaryModel):
+            return PrimaryModelBulkEditForm
+        if issubclass(model, OrganizationalModel):
+            return OrganizationalModelBulkEditForm
+        if issubclass(model, NestedGroupModel):
+            return NestedGroupModelBulkEditForm
+        if issubclass(model, NetBoxModel):
+            return NetBoxModelBulkEditForm
+
+    @staticmethod
+    def get_import_form_base_class(model):
+        """
+        Return the base form class for importing the given model.
+        """
+        if model._meta.app_label == 'dummy_plugin':
+            return
+        if issubclass(model, PrimaryModel):
+            return PrimaryModelImportForm
+        if issubclass(model, OrganizationalModel):
+            return OrganizationalModelImportForm
+        if issubclass(model, NestedGroupModel):
+            return NestedGroupModelImportForm
+        if issubclass(model, NetBoxModel):
+            return NetBoxModelImportForm
+
+    @staticmethod
+    def get_filterset_form_base_class(model):
+        """
+        Return the base form class for the given model's FilterSet.
+        """
+        if model._meta.app_label == 'dummy_plugin':
+            return
+        if issubclass(model, PrimaryModel):
+            return PrimaryModelFilterSetForm
+        if issubclass(model, OrganizationalModel):
+            return OrganizationalModelFilterSetForm
+        if issubclass(model, NestedGroupModel):
+            return NestedGroupModelFilterSetForm
+        if issubclass(model, NetBoxModel):
+            return NetBoxModelFilterSetForm
+
+    def test_model_form_base_classes(self):
+        """
+        Check that each model form inherits from the appropriate base class.
+        """
+        for model in apps.get_models():
```

Comment:
This might need to be tweaked, there are a couple model forms that don't show the edit field.